scene/gui/popup_menu.cpp

```diff
@@ -2967,7 +2967,15 @@ void PopupMenu::popup(const Rect2i &p_bounds) {
 #endif
 
 	if (native) {
-		NativeMenu::get_singleton()->popup(global_menu, (p_bounds != Rect2i()) ? p_bounds.position : get_position());
+		Point2i popup_pos = (p_bounds != Rect2i()) ? p_bounds.position : get_position();
+		if (is_embedded()) {
+			popup_pos = get_embedder()->get_screen_transform().xform(popup_pos); // Note: for embedded windows "screen transform" is transform relative to embedder not the actual screen.
+			DisplayServer::WindowID wid = get_window_id() != DisplayServer::INVALID_WINDOW_ID ? get_window_id() : DisplayServer::MAIN_WINDOW_ID;
```

Comment:
`get_window_id()` isn't exactly trivial, so it would be better to avoid needlessly calling it twice.